src/test/java/org/icgc/argo/program_service/services/EgoServiceIT.java

```diff
@@ -0,0 +1,23 @@
+package org.icgc.argo.program_service.services;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+@ActiveProfiles({ "test", "default" })
+class EgoServiceIT {
+  @Autowired
+  EgoService egoService;
+
+  @Test
+  void egoServiceInitialization() {
+    assertThat(ReflectionTestUtils.getField(egoService, "credentialsProvider")).isNotNull();
```

Comment:
It's better to make a getter because of business logic, not just for the sake of testing. CredentialsProvider is meant only be accessed by ego service, making a getter for it would cause other developers abuse it.